### DISCUSSION TITLE
Render links as vue components

### DISCUSSION THREAD

**Discussion by some-user123** [imzbf/md-editor-v3#232] [Q&A]
I'm using md-editor-v3 in nuxt/vue.

To make use of the fantastic power of vue and nuxt, I would like to render links as `<router-link>` or `<nuxt-link>` components.

How would that be possible?

I know that I can configure the renderer like this:
```ts
MdEditor.config({
  markedRenderer(renderer) {
    renderer.link = (href, title, text) => {
      return `<a href="${href}" title="${title}" target="_blank">${text}</a>`
    }

    return renderer
  }
})
```

But I don't know, how I can put in a vue component like `<router-link>` here... Any ideas?

**Comment by imzbf** [comment]
I have an idea: 

```vue
<template>
  <div>Some Words</div>
</template>

<script setup>
const router = useRouter();

MdEditor.config({
  markedRenderer(renderer) {
    renderer.link = (href, title, text) => {
      return `<a block-jump data-href="${href}" title="${title}" target="_blank">${text}</a>`
    }

    return renderer
  }
})

const clickHandler = (href) => {
 router.push({ path: href });
}

onMounted(() => {
  document.querySelecterAll('a[block-jump ]').forEach(item => {
    item.onclick =  (e) => {
      clickHandler(e.target.dataSet.href);
      
      return false;
    }
  })
})
</script>
```

**Reply by some-user123** [reply]
Thanks a lot for your suggestion!

`e.target.dataSet` does not seem to exist. `e.target.dataset` works.

Would you agree, this is rather a workaround? Because it does not allow to to leverage the full power of vue components... (e.g. create a custom link component)

**Reply by imzbf** [reply]
Yes, we can't use the features of vue components. More biased towards html strings.
